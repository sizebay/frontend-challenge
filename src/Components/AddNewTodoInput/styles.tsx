import styled from "styled-components";
import { theme } from "../../GlobaStyles/theme";

export const InputContainer = styled.form`
  width: 100%;
  display: flex;
  height: 48px;

    input{ 
        width: 100%;
        border: 1px solid ${theme.colors.gray200} ;
        border-radius: 4px 0 0 4px;
        padding-left: 20px;

        ::placeholder{
            color: ${theme.colors.gray500};
            opacity: 0.5;
        }
    }
    button{
        display: grid;
        place-content: center;
        place-items: center;
        border: none;
        width: 44px;
        background-color: ${theme.colors.cyan500};
        border-radius: 0 4px 4px 0;
        transition: opacity 0.2s ease-in-out;
    }
`;
