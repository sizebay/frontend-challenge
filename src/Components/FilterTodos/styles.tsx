import styled from 'styled-components';
import { theme } from '../../GlobaStyles/theme';

interface LabelTypes {
    selected: boolean
}

export const FilterTodosContainer = styled.div`
  display: flex;
  gap: 3.3rem;
  justify-content: space-between;
  margin-bottom: 1.5rem;

  @media(max-width: 800px) {
    flex-direction: column-reverse;
    gap: .9rem;
    margin-bottom: 1rem;
  }
`;

export const FilterButtonsContainer = styled.div`
    display: flex;
    align-items: center;
    justify-content: center;
    gap: 1rem;


`

export const TagButton = styled.button<LabelTypes>`
    position: relative;
    border: 1px solid ${theme.colors.gray200};
    padding: .5rem 1rem;
    border-radius: 300px;
    transition: all 0.2s ease-in-out;

    ${props => props.selected && `
        border: 1px solid ${theme.colors.cyan500};
        color ${theme.colors.cyan500};
        background-color: ${theme.colors.gray100};
        ::before{
            content: "âœ“";

    `}

`
export const FilterInputContainer = styled.div`
    display: flex;
    align-items: center;
    justify-content: center;
    border: 1px solid ${theme.colors.gray200};
    width: 100%;
`

export const FilterInput = styled.input`
    width: 100%;
    height: 40px;
    outline: none;
    padding-left: 1rem;
    border: none;
`

