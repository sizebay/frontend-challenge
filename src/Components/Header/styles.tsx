import styled from 'styled-components';
import { theme } from '../../GlobaStyles/theme';

export const HeaderContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 100%;
  margin-bottom: 24px;
`;

export const DateHolder = styled.div`
  display: grid;
  justify-content: center;
  grid-template-areas:
    "DateNumber DateMonth"
    "DateNumber DateYear";
`;

export const DateNumber = styled.p`
  grid-area: DateNumber;
  font-size: ${theme.font.size[60]};
  font-weight: ${theme.font.weight.medium};
`;

export const DateMonth = styled.span`
  grid-area: DateMonth;
  align-self: end;
  font-size: ${theme.font.size[24]};
  font-weight: ${theme.font.weight.regular};
  width: 3ch;
  white-space: nowrap;
  overflow: hidden;
`;

export const DateYear = styled.span`
  grid-area: DateYear;
  font-size: ${theme.font.size[24]};
  font-weight: ${theme.font.weight.light};
`;

export const WeekDay = styled.p`
  font-size: ${theme.font.size[24]};
  font-weight: ${theme.font.weight.regular};
`;
