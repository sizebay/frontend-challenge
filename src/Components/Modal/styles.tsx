import styled from 'styled-components';
import { theme } from '../../GlobaStyles/theme';

export const ModalContainer = styled.main`
  width: 100vw;
  max-width: 800px;
  height: 650px;
  background-color: ${theme.colors.white};
  border-radius: 4px;
  padding: 54px 60px;

  
  @media(max-width: 800px) {
    height: 90vh;
    max-width: 90vw;
  }
`;
