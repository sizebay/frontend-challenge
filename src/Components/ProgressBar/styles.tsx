import styled from 'styled-components';
import { theme } from '../../GlobaStyles/theme';

interface ProgressType {
    percentage: number;
}

export const ProgressBarContainer = styled.div`
  height: 24px;
  width: 100%;
  background-color: ${theme.colors.gray100};
  border-radius: 4px;
  margin-bottom: 24px;
  overflow: hidden;
`;

export const Progress = styled.div<ProgressType>`
  width: ${props => props.percentage}%;
  background-color: ${theme.colors.green500};
  height: 100%;

  transition: width 0.2s ease-in-out;
`;

