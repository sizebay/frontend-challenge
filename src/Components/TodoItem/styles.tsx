import styled from "styled-components";
import { theme } from "../../GlobaStyles/theme";

interface TodoContainerType {
  completed: boolean
}

export const TodoContainer = styled.div<TodoContainerType>`
  position: relative;
  width: 100%;
  min-height: 48px;
  background-color: ${theme.colors.gray100};
  border: 1px solid ${theme.colors.gray200};
  border-radius: 4px;
  display: flex;
  align-items: center;
  padding-left: 1rem;
  justify-content: space-between;

  ${props => props.completed && `
    opacity: 0.5;
    transition: opacity .2s ease-in-out;
    z-index: 999;
  `}
`;

export const Tooltip = styled.div`
  position: absolute;
  top: 50%;
  left: 50%;
  background-color: ${theme.colors.gray100};
  border: 1px solid ${theme.colors.gray200};
  padding: .5rem;
  display: flex;
  align-items: center;
  justify-content: space-between;
  font-size: .8rem;
  font-weight: bold;
  transition: opacity .2s ease-in-out;
  pointer-events: none;

  button{
    background-color: transparent;
    border: none;
    pointer-events: all;
  }
`

export const ActionButtons = styled.div`
  width: 88px;
  position: relative;
  display: flex;
  align-items: center;
  justify-content: flex-end;
  width: fit-content;
  height: 100%;

  i{
    position: absolute;
    right: 0;
    display: flex;
    align-items: center;
    justify-content: center;
    width: 50%;
    height: 100%;
    background-color: ${theme.colors.green500};
    pointer-events: none;
    border-radius: 0 4px 4px 0
  }
`

export const DeleteButton = styled.button`
  height: 100%;
  width: 44px;
  background-color: ${theme.colors.red500};
  border: none;
`

export const CompleteButton = styled.input`
  height: 100%;
  width: 44px;
  background-color: ${theme.colors.green500};
  border: none;
`

export const EditInput = styled.input`
  height: 44px;
  width: 100%;
  border-radius: 4px;
  border: none;
`

export const EditControls = styled.div`
  position: absolute;
  bottom: -40px;
  left: 50%;
  right: 50%;
  transform: translateX(-50%);
  width: 86px;
  height: 50px;
  display: flex;
  z-index: 10;
  clip-path: polygon(55% 15%, 100% 15%, 100% 100%, 0 100%, 0 15%, 45% 15%, 50% 0);

  button{
    padding-top: 5px;
    width: 100%;
    border: none;
    background-color: ${theme.colors.gray500};
    color: ${theme.colors.white};
    font-size: ${theme.font.size[14]};
    font-size: ${theme.font.weight.regular};
  }

`

export const TodoText = styled.p`
  width: calc(100% - 2rem);
  height: 100%;
  display: flex;
  align-items: center;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
`

