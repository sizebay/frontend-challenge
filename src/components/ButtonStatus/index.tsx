import React from 'react';
import { StyledButton, IconContainer } from './style';

interface ButtonStatusProps {
    isSelected: boolean;
    onClick: () => void;
    icon: React.ReactNode;
    label: string;
  }
  
  const ButtonStatus: React.FC<ButtonStatusProps> = ({ isSelected, onClick, icon, label }) => {
    return (
      <StyledButton isSelected={isSelected} onClick={onClick}>
        <IconContainer isSelected={isSelected}>
          {icon}
        </IconContainer>
        {label}
      </StyledButton>
    );
  };
  
  export default ButtonStatus;