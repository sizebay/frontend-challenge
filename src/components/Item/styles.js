import styled from 'styled-components';

import minusImg from '../../assets/minus.svg';
import checkImg from '../../assets/check.svg';

export const ItemContainer = styled.form`
  position: relative;
  display: flex;
  border: 1px solid #DBDBDB;
  border-radius: 4px;
  background: #F4F4F4;
  color: #848484;
  margin-bottom: 1rem;
  margin-right: 10px;
  height: 3.5rem;

  input {
    width: 100%;
    background: none;
    color: #848484;
    font-size: 1rem;
    border: none;
    padding-left: 1rem;

    &:focus {
      outline: none;
      border: none;
    }
  }

  &:focus-within > button {
    display: inline-block;
  }

  &:focus-within > input {
    background: #fff 0% 0% no-repeat padding-box;
    border-radius: 0.4rem 0 0 0.4rem;
    position: relative;
  }
`;

export const ItemContainerNoEdit = styled.form`
  display: flex;
  border: 1px solid #DBDBDB;
  border-radius: 4px;
  background: #F4F4F4;
  color: #848484;
  margin-bottom: 1rem;
  margin-right: 10px;
  height: 3.5rem;

  input {
    width: 100%;
    background: none;
    color: #848484;
    font-size: 1rem;
    border: none;
    padding-left: 1rem;
    cursor: not-allowed;

    &:focus {
      outline: none;
      border: none;
    }
  }

  &:focus-within > button {
    display: inline-block;
  }

  &:focus-within > input {
    border-radius: 0.4rem 0 0 0.4rem;
    position: relative;
  }
`;

export const RemoveButton = styled.button`
  display: ${(props) => (props.show ? 'block' : 'none')};
  width: 3rem;

  cursor: pointer;
  border: transparent;
  outline: transparent;
  background: #e34f4f 0% 0% no-repeat padding-box;

  &::before {
    content: "";
    background-image: url(${minusImg});
    background-color: rgb(255, 255, 255);
    background-repeat: no-repeat;
    background-position: center;
    display: inline-block;
    width: 1.34rem;
    height: 1.34rem;
    border-radius: 50%;
  }
`;

export const DoneButton = styled.button`
  display: ${(props) => (props.show ? 'block' : 'none')};
  flex-direction: center;
  align-items: center;
  width: 3rem;
  cursor: pointer;
  border: transparent;
  outline: transparent;
  background: #5de290 0% 0% no-repeat padding-box;

  &::before {
    content: "";
    background-image: url(${checkImg});
    background-color: rgb(255, 255, 255);
    background-repeat: no-repeat;
    background-position: center;
    display: inline-block;
    width: 1.34rem;
    height: 1.34rem;
    border-radius: 50%;
  }
`;

export const ItemTooltip = styled.div`
  position: absolute;
  display: flex;
  justify-content: center;
  align-items: center;
  height: 5.34vh;
  width: 4.47vw;
  background: #848484;
  color: #fff;
  top: 80%;
  left: 40%;
  border-radius: 4px;
  z-index: 1;
  padding-left: 0.5rem;
  padding-right: 0.5rem;

  -webkit-touch-callout: none;
  -webkit-user-select: none;
  -khtml-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;

  &::after {
    content: "";
    position: absolute;
    top: -18%;
    left: 50%;
    margin-left: -5px;
    border-width: 5px;
    border-style: solid;
    border-color: #848484 transparent transparent transparent;
    transform: rotate(180deg);
  }

  @media (max-width: 768px) {
    display: none;
  }
`;
