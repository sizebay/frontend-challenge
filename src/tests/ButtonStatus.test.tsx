import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import "@testing-library/jest-dom";
import { theme } from "../styles/theme";
import ButtonStatus from "../components/ButtonStatus/index";

describe("ButtonStatus component", () => {
  it("renders correctly with default props", () => {
    const { getByText } = render(
      <ThemeProvider theme={theme}>
        <ButtonStatus
          isSelected={false}
          onClick={() => {}}
          icon={<div>Icon</div>}
          label="Click me"
        />
      </ThemeProvider>
    );

    expect(getByText("Click me"));
    expect(getByText("Icon"));
    expect(document.body).toMatchSnapshot();
  });

  it("calls onClick handler when clicked", () => {
    const onClickMock = jest.fn();
    const { getByText } = render(
      <ThemeProvider theme={theme}>
        <ButtonStatus
          isSelected={false}
          onClick={onClickMock}
          icon={<div>Icon</div>}
          label="Click me"
        />
      </ThemeProvider>
    );

    fireEvent.click(getByText("Click me"));

    expect(onClickMock).toHaveBeenCalledTimes(1);
    expect(document.body).toMatchSnapshot();
  });
});
