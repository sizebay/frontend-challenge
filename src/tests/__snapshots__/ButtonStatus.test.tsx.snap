// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonStatus component calls onClick handler when clicked 1`] = `
<body>
  <div>
    <button
      class="sc-gEvEer ipests"
    >
      <span
        class="sc-aXZVg gfMXTp"
      >
        <div>
          Icon
        </div>
      </span>
      Click me
    </button>
  </div>
</body>
`;

exports[`ButtonStatus component renders correctly with default props 1`] = `
<body>
  <div>
    <button
      class="sc-gEvEer ipests"
    >
      <span
        class="sc-aXZVg gfMXTp"
      >
        <div>
          Icon
        </div>
      </span>
      Click me
    </button>
  </div>
</body>
`;
